### PR TITLE
Add support for new fields in external services template

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Enhancement / new feature

### Description

In some cases, it might be useful to be able to edit the `externalTrafficPolicy` and `loadBalancerSourceRanges` options of the load balancer or node port services. Their value might be in:
* Propagating the real source IP address into the Kafka brokers (`externalTrafficPolicy`)
* Making sure the routing is more efficient by not routing node port traffic between nodes not hosting the endpoint locally (`externalTrafficPolicy`)
* Configure the cloud provider firewalls to allow connections only from some IPs (`loadBalancerSourceRanges`)

### Checklist

- [x] Write tests
- [x] Make sure all tests pass
- [x] Update documentation
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [x] Update CHANGELOG.md

### CODE REVIEW

File: api/src/main/java/io/strimzi/api/kafka/model/template/ExternalServiceTemplate.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+ * It contains additional values applicable to load balancer or node port type services.
+ */
+@Buildable(
+        editableEnabled = false,
+        generateBuilderPackage = false,
+        builderPackage = "io.fabric8.kubernetes.api.builder"
+)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonPropertyOrder({
+        "metadata"})
```
Comment (ppatierno) [inline api/src/main/java/io/strimzi/api/kafka/model/template/ExternalServiceTemplate.java]: missing "externalTrafficPolicy" and "loadBalancerSourceRanges"

File: documentation/modules/con-customizing-external-services.adoc [hunk 2]
```diff
@@ -0,0 +1,51 @@
+// This assembly is included in the following assemblies:
+//
+// assembly-customizing-deployments.adoc
+
+[id='con-customizing-external-services-{context}']
+= Customizing external Services
+
+When exposing Kafka to the outside of Kubernetes using load balancers or node prots, you can in addition to Labels and Annotations customize some more fields .
```
Comment (ppatierno) [inline documentation/modules/con-customizing-external-services.adoc]: typo "prots".
"you can customize some more fields in addition to Labels and Annotations" sounds better to me, but it's more for @PaulRMellor or @laidan6000

File: api/src/main/java/io/strimzi/api/kafka/model/template/ExternalServiceTemplate.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    public MetadataTemplate getMetadata() {
+        return metadata;
+    }
+
+    public void setMetadata(MetadataTemplate metadata) {
+        this.metadata = metadata;
+    }
+
+    @Description("Denotes if this Service desires to route external traffic to node-local or cluster-wide endpoints. " +
```
Comment (PaulRMellor) [inline api/src/main/java/io/strimzi/api/kafka/model/template/ExternalServiceTemplate.java]:
```suggestion
    @Description("Specifies whether the service routes external traffic to node-local or cluster-wide endpoints. " +
```

File: api/src/main/java/io/strimzi/api/kafka/model/template/ExternalServiceTemplate.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+            "`Cluster` may cause a second hop to another node and obscures the client source IP. " +
+            "`Local` avoids a second hop for LoadBalancer and Nodeport type services and preserves the client source IP (when supported by the infrastructure). " +
+            "When not specified, Kubernetes will use `Cluster` as default.")
```
Comment (PaulRMellor) [inline api/src/main/java/io/strimzi/api/kafka/model/template/ExternalServiceTemplate.java]:
```suggestion
            "If unspecified, Kubernetes will use `Cluster` as the default.")
```

File: api/src/main/java/io/strimzi/api/kafka/model/template/ExternalServiceTemplate.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    public ExternalTrafficPolicy getExternalTrafficPolicy() {
+        return externalTrafficPolicy;
+    }
+
+    public void setExternalTrafficPolicy(ExternalTrafficPolicy externalTrafficPolicy) {
+        this.externalTrafficPolicy = externalTrafficPolicy;
+    }
+
+    @Description("A list of CIDR ranges (for example `10.0.0.0/8` or `130.211.204.1/32`) from which clients will be able to connect to load balancer type listeners. " +
```
Comment (PaulRMellor) [inline api/src/main/java/io/strimzi/api/kafka/model/template/ExternalServiceTemplate.java]:
```suggestion
    @Description("A list of CIDR ranges (for example `10.0.0.0/8` or `130.211.204.1/32`) from which clients can connect to load balancer type listeners. " +
```

File: api/src/main/java/io/strimzi/api/kafka/model/template/ExternalServiceTemplate.java [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+            "If supported by the platform, this will restrict traffic through the load balancer to the specified CIDR ranges in this list. " +
```
Comment (PaulRMellor) [inline api/src/main/java/io/strimzi/api/kafka/model/template/ExternalServiceTemplate.java]:
```suggestion
            "If supported by the platform, traffic through the load balancer is restricted to the specified CIDR ranges. " +
```

File: api/src/main/java/io/strimzi/api/kafka/model/template/ExternalServiceTemplate.java [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+            "This field is applicable only for load balancer type services and will be ignored if the cloud-provider does not support the feature. " +
```
Comment (PaulRMellor) [inline api/src/main/java/io/strimzi/api/kafka/model/template/ExternalServiceTemplate.java]:
```suggestion
            "This field is applicable only for loadbalancer type services and is ignored if the cloud provider does not support the feature. " +
```

File: api/src/main/java/io/strimzi/api/kafka/model/template/ExternalServiceTemplate.java [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+            "For more info see https://kubernetes.io/docs/tasks/access-application-cluster/configure-cloud-provider-firewall/")
```
Comment (PaulRMellor) [inline api/src/main/java/io/strimzi/api/kafka/model/template/ExternalServiceTemplate.java]:
```suggestion
            "For more information, see https://kubernetes.io/docs/tasks/access-application-cluster/configure-cloud-provider-firewall/")
```

File: documentation/modules/appendix_crds.adoc [hunk 9]
```diff
@@ -955,6 +955,23 @@ Used in: xref:type-EntityOperatorTemplate-{context}[`EntityOperatorTemplate`], x
...
+
+Used in: xref:type-KafkaClusterTemplate-{context}[`KafkaClusterTemplate`]
+
+
+[options="header"]
+|====
+|Property                         |Description
+|metadata                  1.2+<.<|Metadata which should be applied to the resource.
+|xref:type-MetadataTemplate-{context}[`MetadataTemplate`]
+|externalTrafficPolicy     1.2+<.<|Denotes if this Service desires to route external traffic to node-local or cluster-wide endpoints. `Cluster` may cause a second hop to another node and obscures the client source IP. `Local` avoids a second hop for LoadBalancer and Nodeport type services and preserves the client source IP (when supported by the infrastructure). When not specified, Kubernetes will use `Cluster` as default.
```
Comment (PaulRMellor) [inline documentation/modules/appendix_crds.adoc]:
```suggestion
|externalTrafficPolicy     1.2+<.<|Specifies whether the service routes external traffic to node-local or cluster-wide endpoints. `Cluster` may cause a second hop to another node and obscures the client source IP. `Local` avoids a second hop for LoadBalancer and Nodeport type services and preserves the client source IP (when supported by the infrastructure). If unspecified, Kubernetes will use `Cluster` as the default.
```

File: documentation/modules/appendix_crds.adoc [hunk 10]
```diff
@@ -955,6 +955,23 @@ Used in: xref:type-EntityOperatorTemplate-{context}[`EntityOperatorTemplate`], x
...
+|string (one of [Local, Cluster])
+|loadBalancerSourceRanges  1.2+<.<|A list of CIDR ranges (for example `10.0.0.0/8` or `130.211.204.1/32`) from which clients will be able to connect to load balancer type listeners. If supported by the platform, this will restrict traffic through the load balancer to the specified CIDR ranges in this list. This field is applicable only for load balancer type services and will be ignored if the cloud-provider does not support the feature. For more info see https://kubernetes.io/docs/tasks/access-application-cluster/configure-cloud-provider-firewall/.
```
Comment (PaulRMellor) [inline documentation/modules/appendix_crds.adoc]:
```suggestion
|loadBalancerSourceRanges  1.2+<.<|A list of CIDR ranges (for example `10.0.0.0/8` or `130.211.204.1/32`) from which clients can connect to load balancer type listeners. If supported by the platform, traffic through the loadbalancer is restricted to the specified CIDR ranges. This field is applicable only for loadbalancer type services, and is ignored if the cloud provider does not support the feature. For more information, see https://kubernetes.io/docs/tasks/access-application-cluster/configure-cloud-provider-firewall/.
```

File: documentation/modules/con-customizing-external-services.adoc [hunk 11]
```diff
@@ -0,0 +1,51 @@
+// This assembly is included in the following assemblies:
+//
+// assembly-customizing-deployments.adoc
+
+[id='con-customizing-external-services-{context}']
+= Customizing external Services
+
+When exposing Kafka to the outside of Kubernetes using load balancers or node ports, you can customize some more fields in addition to Labels and Annotations.
```
Comment (PaulRMellor) [inline documentation/modules/con-customizing-external-services.adoc]:
```suggestion
When exposing Kafka outside of Kubernetes using loadbalancers or node ports, you can use additional customization properties in addition to labels and annotations.
```

File: documentation/modules/con-customizing-external-services.adoc [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+These fields are described in the following table and affect how the Service is created.
```
Comment (PaulRMellor) [inline documentation/modules/con-customizing-external-services.adoc]:
```suggestion
The properties for external services are described in the following table and affect how the service is created.
```

File: documentation/modules/con-customizing-external-services.adoc [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+
+[table,stripes=none]
+|===
+|Field |Description
+
+|`externalTrafficPolicy`
+|Denotes if this Service desires to route external traffic to node-local or cluster-wide endpoints.
```
Comment (PaulRMellor) [inline documentation/modules/con-customizing-external-services.adoc]:
```suggestion
|Specifies whether the service routes external traffic to node-local or cluster-wide endpoints.
```

File: documentation/modules/con-customizing-external-services.adoc [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+`Cluster` may cause a second hop to another node and obscures the client source IP.
+`Local` avoids a second hop for LoadBalancer and Nodeport type services and preserves the client source IP (when supported by the infrastructure).
+When not specified, Kubernetes will use `Cluster` as default.
```
Comment (PaulRMellor) [inline documentation/modules/con-customizing-external-services.adoc]:
```suggestion
If unspecified, Kubernetes will use `Cluster` as the default.
```

File: documentation/modules/con-customizing-external-services.adoc [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+
+|`loadBalancerSourceRanges`
+|A list of CIDR ranges (for example `10.0.0.0/8` or `130.211.204.1/32`) from which clients will be able to connect to load balancer type listeners.
```
Comment (PaulRMellor) [inline documentation/modules/con-customizing-external-services.adoc]:
```suggestion
|A list of CIDR ranges (for example `10.0.0.0/8` or `130.211.204.1/32`) from which clients can connect to load balancer type listeners.
```

File: documentation/modules/con-customizing-external-services.adoc [hunk 16]
```diff
@@ -0,0 +1,51 @@
...
+If supported by the platform, this will restrict traffic through the load balancer to the specified CIDR ranges in this list.
```
Comment (PaulRMellor) [inline documentation/modules/con-customizing-external-services.adoc]:
```suggestion
If supported by the platform, traffic through the loadbalancer is restricted to the specified CIDR ranges.
```

File: documentation/modules/con-customizing-external-services.adoc [hunk 17]
```diff
@@ -0,0 +1,51 @@
...
+This field is applicable only for load balancer type services and will be ignored if the cloud-provider does not support the feature.
```
Comment (PaulRMellor) [inline documentation/modules/con-customizing-external-services.adoc]:
```suggestion
This field is applicable only for loadbalancer type services, and is ignored if the cloud provider does not support the feature.
```

File: documentation/modules/con-customizing-external-services.adoc [hunk 18]
```diff
@@ -0,0 +1,51 @@
...
+
+For more info see link:https://kubernetes.io/docs/tasks/access-application-cluster/configure-cloud-provider-firewall/[Configure Your Cloud Provider's Firewalls].
```
Comment (PaulRMellor) [inline documentation/modules/con-customizing-external-services.adoc]:
```suggestion
For more information, see https://kubernetes.io/docs/tasks/access-application-cluster/configure-cloud-provider-firewall/.
```

File: documentation/modules/con-customizing-external-services.adoc [hunk 19]
```diff
@@ -0,0 +1,51 @@
...
+|===
+
+These fields are available on the `externalBootstrapService` and `perPodService` fields.
```
Comment (PaulRMellor) [inline documentation/modules/con-customizing-external-services.adoc]:
```suggestion
These properties are available for `externalBootstrapService` and `perPodService`.
```

File: documentation/modules/con-customizing-external-services.adoc [hunk 20]
```diff
@@ -0,0 +1,51 @@
...
+The following example shows these customized fields on a `template` property:
```
Comment (PaulRMellor) [inline documentation/modules/con-customizing-external-services.adoc]:
```suggestion
The following example shows these customized properties for a `template`:
```

**Pull request merged**